app/ide-desktop/lib/dashboard/src/modals/UpsertSecretModal.tsx

```diff
@@ -49,56 +54,79 @@ export default function UpsertSecretModal(props: UpsertSecretModalProps) {
         data-testid="upsert-secret-modal"
         tabIndex={-1}
         className="pointer-events-auto relative flex w-upsert-secret-modal flex-col gap-modal rounded-default p-modal-wide pt-modal before:absolute before:inset before:h-full before:w-full before:rounded-default before:bg-selected-frame before:backdrop-blur-default"
-        onKeyDown={event => {
-          if (event.key !== 'Escape') {
-            event.stopPropagation()
-          }
-        }}
         onClick={event => {
           event.stopPropagation()
         }}
         onSubmit={event => {
           event.preventDefault()
-          onSubmit()
+          doSubmit()
         }}
       >
-        <h1 className="relative text-sm font-semibold">
+        <aria.Heading level={2} className="relative text-sm font-semibold">
           {isCreatingSecret ? getText('newSecret') : getText('editSecret')}
-        </h1>
-        <label className="relative flex h-row items-center">
-          <div className="text w-modal-label">{getText('name')}</div>
-          <input
-            autoFocus
-            disabled={!isNameEditable}
-            placeholder={getText('secretNamePlaceholder')}
-            className="text grow rounded-full border border-primary/10 bg-transparent px-input-x selectable enabled:active"
-            value={name}
-            onInput={event => {
-              setName(event.currentTarget.value)
-            }}
-          />
-        </label>
-        <label className="relative flex h-row items-center">
-          <div className="text w-modal-label">{getText('value')}</div>
-          <input
-            autoFocus={!isNameEditable}
-            placeholder={
-              isNameEditable ? getText('secretValuePlaceholder') : getText('secretValueHidden')
-            }
-            className="text grow rounded-full border border-primary/10 bg-transparent px-input-x"
-            onInput={event => {
-              setValue(event.currentTarget.value)
-            }}
-          />
-        </label>
-        <div className="relative flex gap-buttons">
-          <button disabled={!canSubmit} type="submit" className="button bg-invite text-white">
+        </aria.Heading>
+        <div className="relative flex flex-col">
+          <FocusArea direction="horizontal">
+            {(ref, innerProps) => (
+              <aria.TextField
+                ref={ref}
+                className="relative flex h-row items-center"
+                {...innerProps}
+              >
+                <aria.Label className="text w-modal-label">{getText('name')}</aria.Label>
+                <FocusRing>
+                  <aria.Input
+                    autoFocus
+                    disabled={!isNameEditable}
+                    placeholder={getText('secretNamePlaceholder')}
+                    className="focus-child text grow rounded-full border border-primary/10 bg-transparent px-input-x selectable enabled:active"
```

Comment:
I'm not sure it's a good idea to use implicit contract to make an element focusable. I think we should hide this behind a hook and use the [mergeProps](https://react-spectrum.adobe.com/react-aria/mergeProps.html) function to apply this behavior.